simulation/halsim_ds_socket/src/main/native/cpp/DSCommPacket.cpp

```diff
@@ -239,7 +239,8 @@ void DSCommPacket::ReadJoystickDescriptionTag(wpi::span<const uint8_t> data) {
   packet.descriptor.name[nameLength] = '\0';
   int axesCount = data[0];
   packet.descriptor.axisCount = axesCount;
-  for (int i = 0; i < axesCount; i++) {
+  for (int i = 0;
+       i < std::min<int>(axesCount, sizeof(packet.descriptor.axisTypes)); i++) {
```

Comment:
Without optimization, this generates much better code:
```
  for (int i = 0, len = std::min<int>(axesCount, sizeof(packet.descriptor.axisTypes));
       i < len; i++)
```

With -O this compiles identical code.